src/mpi/init/initthread.c

```diff
@@ -178,31 +168,7 @@ int MPIR_Init_thread(int *argc, char ***argv, int required, int *provided)
     MPID_THREAD_CS_ENTER(GLOBAL, MPIR_THREAD_GLOBAL_ALLFUNC_MUTEX);
     exit_init_cs_on_failure = 1;
 
-#ifdef HAVE_HWLOC
-    MPIR_Process.bindset = hwloc_bitmap_alloc();
-    hwloc_topology_init(&MPIR_Process.hwloc_topology);
-    MPIR_Process.bindset_is_valid = 0;
-    hwloc_topology_set_io_types_filter(MPIR_Process.hwloc_topology, HWLOC_TYPE_FILTER_KEEP_ALL);
-    if (!hwloc_topology_load(MPIR_Process.hwloc_topology)) {
-        MPIR_Process.bindset_is_valid =
-            !hwloc_get_proc_cpubind(MPIR_Process.hwloc_topology, getpid(), MPIR_Process.bindset,
-                                    HWLOC_CPUBIND_PROCESS);
-    }
-#endif
-
-#ifdef HAVE_NETLOC
-    MPIR_Process.network_attr.u.tree.node_levels = NULL;
-    MPIR_Process.network_attr.network_endpoint = NULL;
-    MPIR_Process.netloc_topology = NULL;
-    MPIR_Process.network_attr.type = MPIR_NETLOC_NETWORK_TYPE__INVALID;
-    if (strlen(MPIR_CVAR_NETLOC_NODE_FILE)) {
-        mpi_errno =
-            netloc_parse_topology(&MPIR_Process.netloc_topology, MPIR_CVAR_NETLOC_NODE_FILE);
-        if (mpi_errno == NETLOC_SUCCESS) {
-            MPIR_Netloc_parse_topology(MPIR_Process.netloc_topology, &MPIR_Process.network_attr);
-        }
-    }
-#endif
+    MPIDU_topo_init();
```

Comment:
Please be careful with topology code because I want to actually replace it with this https://github.com/pmodels/mpich/pull/3594 as soon as `topotree` code is refactored.

Reply:
I will be careful:).

Actually, I am just tucking the code into a "box" and moving the box into a topo subtree.  I am not going to touch the actual "topo" code.

Reply:
What I meant is that you won't need to touch this at all as I hope I can merge my PR before you finish yours. If you finish first, you may not want to touch this either as my PR will wipe out your changes anyway.